i18n/locales/en.ts

```diff
@@ -1233,21 +1241,20 @@ const translations = {
         'Segregated Witness transactions work just like their legacy predecessors. There are a few global values like version and locktime. There is an array of inputs (UTXOs we want to spend) and an array of outputs (new UTXOs we want to create, for other people to spend in the future). There will also be an array of witnesses, one for each input. That is where signatures and scripts will go instead of the scriptSig.',
       paragraph_two:
         'The message serializations for all these components is documented <Link href="https://en.bitcoin.it/wiki/Protocol_documentation#tx" target="_blank" className="underline">here</Link>  and <Link href="https://github.com/bitcoinbook/bitcoinbook/blob/6d1c26e1640ae32b28389d5ae4caf1214c2be7db/ch06_transactions.adoc" target="_black" className="underline" >here</Link>.',
-      paragraph_three:
-        'And remember: integers in Bitcoin are serialized little-endian!',
     },
     in_out_four: {
       title: 'The ins and outs',
       nav_title: 'The input class',
-      heading: 'Looking at the Class Input implementation',
-      paragraph_one: 'It should have the following method:',
+      heading: 'Looking at the Input class implementation',
+      paragraph_one:
+        'Bitcoin transaction inputs always point to existing, unspent transaction outputs. Therefore, our Input class has a method <span className="text-green"> from_output() </span> which is used to construct an Input by passing in the output description:',
       paragraph_two:
-        'The First two arguments are the transaction ID and the index of the output of that transaction you want to spend from.',
+        'The first two arguments are the transaction ID and the index of the output of that transaction you want to spend from. Eventually we will pass in the txid and vout values you got above from listunspent. ',
       paragraph_three:
-        'Eventually we will pass in the txid and vout values you got above from listunspent. Note that hashes in Bitcoin are little-endian, which means that you will need to reverse the byte order of the txid string!',
+        'Note that raw-byte hashes in Bitcoin are reversed, which is why we need to reverse the byte order of the hex-string txid argument that is passed in.',
```

Comment:
```suggestion
        'Note that raw-byte hashes in Bitcoin are reversed (aka little-endian), which is why we need to reverse the byte order of the hex-string txid argument that is passed in.',
```

I'm so sorry, I missed that you had little-endian mentioned here. I think I got so deep into the code portion of the challenge that I forgot to revisit the challenge copy before filing some of my feedback. I really like the improvements you made here but I also agree with the original text that mentioned the term little-endian.

Reply:
SOooooooo... technically "little endian" only accurately describes integers. I asked Pieter what the best term is to describe what we do to hashes and he shrugged out "byte swapped?".

One level deeper, goes like this: if you were to look at the actual output of a sha256 digest for a bitcoin block, you would actually see all the 0's on the *right side*. Satoshi decided to interpret that as a little-endian integer and that is why the int256 values of block hashes gets lower as difficulty increases: more zeros on the right side means a smaller little-endian integer.

Whenever the RPC was added to bitcoin, and for the first time block hashes were printed as strings for humans, someone decided to reverse the bytes of the hash so it looked like a human-readable integer, meaning more zeroes on the *left side* are smaller numbers.

Okay so that's the story as I understand it... how do we get that into the game?

Reply:
Whoa!!!!! That is really cool! That's a fascinating bit of development history that we absolutely have to include in the game! This is going to require some design review but I'm thinking it could be some variation of the following ideas:

- A modal similar to the ChatBTC Holocat icon
- A piece of paper or dropped item that you discover in the game. The player will feel like they found something secret or hidden, but we'll intentionally put it somewhere easy to find :)
- A "History of Bitcoin"/Lore section in the help page

I would love for us to get this into the game sooner than later so that it's at least somewhere, then we can iterate on the best way to display it. I think we should add it to the help page for now. I am going to make note of this on my feedback doc.

As for the code suggestion that started this thread, you were right to remove "little-endian" :D Feel free to resolve this conversation.

Reply:
Ok for now, I expanded on the reverse hash line in the text and added a Holocat tooltip to a longer explanation